### PR TITLE
Replace deprecated httpclient

### PR DESCRIPTION
**Pull request by nermin-yehia**
that review to show you
1-replace deprecated HttpClient.
2-print logs or stop it according to build target

### CODE REVIEW

File: Strike/src/com/starthouse/strike/ltx/LTXService.java [hunk 1]
```diff
@@ -29,12 +29,8 @@
 	private static LTXService ltxDataService;
```
Comment (nermin-yehia) [inline Strike/src/com/starthouse/strike/ltx/LTXService.java]: ltx service responsible for all operations relating to ltx sdk in application

File: Strike/src/com/starthouse/strike/ws/WebServiceClient.java [hunk 2]
```diff
@@ -55,6 +32,12 @@
 import com.starthouse.strike.ws.dto.model.NotificationListDto;
 import com.starthouse.strike.ws.dto.model.RideDto;
 
+/***
+ * this it the web service client class to handle http requests to strike api
+ * 
+ * @author nermin.yehia
+ * 
+ */
 public class WebServiceClient {
```
Comment (nermin-yehia) [inline Strike/src/com/starthouse/strike/ws/WebServiceClient.java]: remove deprecated HttpClient class and user HttpUrlConnection class

File: Strike/src/com/starthouse/strike/ws/WebServiceClient.java [hunk 3]
```diff
@@ -453,37 +198,34 @@ private String getPayloadVariantFromConnection(
...
 
 		String responseString = getPayloadFromConnection(urlConnection);
-
-		Log.d(TAG, "getPayloadVariantFromConnection() payload: "
-				+ responseString);
-
-		Log.d(TAG, "ResponseCode: " + urlConnection.getResponseCode());
-		Log.d(TAG, "Header fields: response " + urlConnection.getHeaderFields());
-
+		if (Config.PRINT_LOGS) {
```
Comment (nermin-yehia) [inline Strike/src/com/starthouse/strike/ws/WebServiceClient.java]: -i added boolean to custom rules to show/hide logs according to build target
-Log has impact on performance, so it's recommended that you comment it out or log with conditional statements.

File: Strike/src/com/starthouse/strike/activity/fragments/AvailableRidesFragment.java [hunk 4]
```diff
@@ -10,6 +10,7 @@
 import com.ltx.sdk.database.tables.Geofence;
 import com.ltx.sdk.database.tables.Node;
 import com.ltx.sdk.database.tables.ProximityGroup;
+import com.starthouse.strike.Config;
 import com.starthouse.strike.database.tables.Ride;
 import com.starthouse.strike.ltx.ListenOnLTXCallback;
 import com.starthouse.strike.model.enums.AvailableRideTypes;
```
Comment (nermin-yehia) [inline Strike/src/com/starthouse/strike/activity/fragments/AvailableRidesFragment.java]: each fragment affected by LTX callbacks it must implement ListenOnLTXCallback

File: Strike/src/com/starthouse/strike/StrikeUtils.java [hunk 5]
```diff
@@ -26,34 +30,41 @@ public static String getRidePickupDescription(Ride ride) {
 		Long rideId = ride.getRideId();
 		if (rideType != null)
 			if (rideType.equals(RideType.POINT_TO_POINT.toString())) {
-				Log.d(TAG,
-						"getRidePickupDescription : " + rideId + " -- type :"
-								+ rideType + " --- object : "
-								+ ride.getRidePointTopoint());
+				if (Config.PRINT_LOGS)
```
Comment (engpeter2010) [inline Strike/src/com/starthouse/strike/StrikeUtils.java]: Actually, I don't know why you did that. You already can control that by increasing logging level of Log

**Pull request merged**